manifests/01-role.yaml

```diff
@@ -36,3 +36,23 @@ rules:
   - services
   verbs:
   - "*"
+---
+# Role for the operator to delete Role and RoleBindings
+# in the openshift-config namespace.
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ingress-operator
+  namespace: openshift-config
+  annotations:
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+rules:
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - roles
+  - rolebindings
+  verbs:
+  - delete
```

Comment:
no creation?

Reply:
The operator already has a clusterRole granting [create, get, list, watch, and update permissions](https://github.com/openshift/cluster-ingress-operator/blob/a5525f387106974730ccb9f422e05115498cbc47/manifests/00-cluster-role.yaml#L51-L63) in all namespaces but I did not want to extend it the ability to delete roles/rolebindings in all namespaces.

Reply:
Would add delete there too. The operator is already basically cluster admin with what it can do (assigning itself more roles). Adding delete is not making it worse.

Wondering for what does it use the rbac permissions today?

Reply:
I agree with your assessment that the operator is effectively a cluster admin and that granting it delete permissions at the cluster level doesn't make anything worse.

However, I believe that by scoping these permissions to the `openshift-config` the purpose of rbac is better communicated to developers unfamiliar with this feature whereas the purpose is somewhat mystified if lumped into the clusterRole. If you do not believe that this information is valuable to other developers I can lump it into the clusterRole.